typescript-selenium/tests/HappyPath.spec.ts

```diff
@@ -0,0 +1,107 @@
+/*********************************************************************
+ * Copyright (c) 2019 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import { e2eContainer } from "../inversify.config";
+import { TYPES, CLASSES } from "../inversify.types";
+import { LoginPage } from "../pageobjects/login/LoginPage";
+import { Dashboard } from "../pageobjects/dashboard/Dashboard";
+import { NameGenerator } from "../utils/NameGenerator";
+import { NewWorkspace } from "../pageobjects/dashboard/NewWorkspace";
+import { WorkspaceDetailsPlugins } from "../pageobjects/dashboard/workspace-details/WorkspaceDetailsPlugins";
+import { Ide } from "../pageobjects/ide/Ide";
+import { ProjectTree } from "../pageobjects/ide/ProjectTree";
+import { Editor } from "../pageobjects/ide/Editor";
+
+const workspaceName: string = NameGenerator.generate("wksp-test-", 5);
+const namespace: string = "che";
+const sampleName: string = "console-java-simple";
+const pluginId: string = "redhat/java/0.38.0";
+const javaPluginName: string = "Language Support for Java(TM)";
+
+const loginPage: LoginPage = e2eContainer.get<LoginPage>(TYPES.LoginPage);
+const dashboard: Dashboard = e2eContainer.get(CLASSES.Dashboard);
+const newWorkspace: NewWorkspace = e2eContainer.get(CLASSES.NewWorkspace);
+const workspaceDetailsPlugins: WorkspaceDetailsPlugins = e2eContainer.get(CLASSES.WorkspaceDetailsPlugins)
+const ide: Ide = e2eContainer.get(CLASSES.Ide)
+const projectTree: ProjectTree = e2eContainer.get(CLASSES.ProjectTree)
+const editor: Editor = e2eContainer.get(CLASSES.Editor)
+
+suite("E2E", async () => {
+
+    suite("Login and wait dashboard", async () => {
+        test("Login", async () => {
+            await loginPage.login()
+        })
+    })
+
+    suite("Create workspace and add plugin", async () => {
+        test("Open 'New Workspace' page", async () => {
+            await newWorkspace.openPageByUI();
+        })
+
+        test(`Create a '${workspaceName}' workspace and proceed editing`, async () => {
+            await newWorkspace.createWorkspaceAndProceedEditing(workspaceName, 'che7-preview', sampleName)
+        })
+
+        test("Add 'Java Language Support' plugin to workspace", async () => {
+            await workspaceDetailsPlugins.addPluginAndOpenWorkspace(namespace, workspaceName, javaPluginName, pluginId)
```

Comment:
Maybe I'll split this logic to two lines: ```addPlugins``` and ```openWorksapce``` because I don't think these operations are atomic and we maybe could want to add more plugins before opening workspace.

Reply:
Yes, you are right, but we can't check that plugins are successfully added until workspace been run.